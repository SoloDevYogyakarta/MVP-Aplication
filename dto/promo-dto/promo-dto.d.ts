export declare class PromoField {
    name: string;
    desc: string;
    price: number;
    discount: number;
    start_time: Date;
    end_time: Date;
}
