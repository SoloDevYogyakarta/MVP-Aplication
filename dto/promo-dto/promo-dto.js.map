{"version":3,"file":"promo-dto.js","sourceRoot":"","sources":["../../../src/dto/promo-dto/promo-dto.ts"],"names":[],"mappings":";;;;;;;;;;;;AAAA,qDAA6D;AAE7D,MAAa,UAAU;CAkBtB;AAlBD,gCAkBC;AAhBC;IADC,IAAA,0BAAQ,GAAE;;wCACE;AAGb;IADC,IAAA,0BAAQ,GAAE;;wCACE;AAGb;IADC,IAAA,0BAAQ,GAAE;;yCACG;AAGd;IADC,IAAA,0BAAQ,GAAE;;4CACM;AAGjB;IADC,IAAA,wBAAM,GAAE;8BACG,IAAI;8CAAC;AAGjB;IADC,IAAA,wBAAM,GAAE;8BACC,IAAI;4CAAC"}