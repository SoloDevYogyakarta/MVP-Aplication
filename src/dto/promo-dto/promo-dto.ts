import { IsDate, IsNumber, IsString } from 'class-validator';

export class PromoField {
  @IsString()
  name: string;

  @IsString()
  desc: string;

  @IsNumber()
  price: number;

  @IsNumber()
  discount: number;

  @IsDate()
  start_time: Date;

  @IsDate()
  end_time: Date;
}
