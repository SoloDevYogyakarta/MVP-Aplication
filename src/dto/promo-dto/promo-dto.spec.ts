import { faker } from '@faker-js/faker';
import { validate } from 'class-validator';
import { PromoField } from './promo-dto';

describe('PromoField', () => {
  it('should to be defined', () => expect(PromoField).toBeDefined());

  it('render correctly', () => expect(PromoField).toMatchSnapshot());

  it('should to be successfully', async () => {
    const data = new PromoField();
    data.name = faker.commerce.productName();
    data.desc = faker.commerce.productDescription();
    data.price = Number(faker.commerce.price());
    data.discount = faker.number.int({ min: 1, max: 100 });
    data.start_time = new Date();
    data.end_time = new Date();
    const valid = await validate(data);
    expect(valid.length).toEqual(0);
  });
});
