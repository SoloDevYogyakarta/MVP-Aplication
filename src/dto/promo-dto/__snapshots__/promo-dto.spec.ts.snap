// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromoField render correctly 1`] = `[Function]`;
